spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/batch/SimpleJobService.java

```diff
@@ -412,7 +418,7 @@ public int countStepExecutionsForJobExecution(long jobExecutionId) {
 
 	@Override
 	public JobInstance getJobInstance(long jobInstanceId) throws NoSuchJobInstanceException {
-		JobInstance jobInstance = jobInstanceDao.getJobInstance(jobInstanceId);
+		JobInstance jobInstance = this.aggregateJobQueryDao.getJobInstance(jobInstanceId, this.schemaTarget);
```

Comment:
The `jobInstanceDao` is support to already be one that was selected from the container.